### ISSUE TITLE
Dom0 Kernel Panic on i7-9750H iGPU

### ISSUE THREAD

**Issue by JarrahG** [QubesOS/qubes-issues#5422]
**Qubes OS version**

R4.0.1, R4.0.2-rc1 and custom built ISOs.

**Affected component(s) or functionality**

Dom0 will not boot on system installer.

**Brief summary**

On boot, Dom0 crashes with a kernel panic on the following hypercall:
```
xen_hypercall_xen_version
```
**To Reproduce**
_Steps to reproduce the behavior:_
_1. Go to '...'_
_2. Click on '....'_
_3. Scroll down to '....'_
_4. See error_
Start Qubes (either ISO or installed) on a laptop with an i7-9750H in EFI mode. On my system, leaving the iGPU enabled results in this behaviour. Disabling the iGPU and enabling only the Dedicated Nvidia card allows booting.

**Expected behavior**

Dom0 boots to installer.
**Actual behavior**

Dom0 kernel panics.
**Screenshots**
![60339282-ddb13380-9997-11e9-89dc-b66d11e8c61d](https://user-images.githubusercontent.com/43260698/67632046-025ddf00-f896-11e9-9012-e6813789a3f5.jpg)

**Additional context**

When set to the dedicated Nvidia GPU, the system is able to boot. However, as the card is not supported by Nouveau, this is not a useable workaround. This does however show that the GPU change is the deciding factor.

I have tested with the following system states:

    ISO/Fresh install: The exact system found the in 4.0.1 ISO.
    Current fully patched QubesOS.

I have been unable to build an ISO to test Kernel-latest. I will be attempting to install on another machine to test whether this solves the problem. 

**Solutions you've tried**

Standard and custom ISOs with different versions (4.0-4.0.2-rc2)

The following boot config was used to gather debug output:
```
[qubes-verbose]
options=console=vga efi=no-rs iommu=no-igfx loglvl=all dom0_mem=min:1024M dom0_mem=max:4096M ucode=scan vga=current,keep guest_loglvl=all
#noexitboot=1
#mapbs=1
kernel=vmlinuz inst.stage2=hd:LABEL=Qubes-20190628-x86_64 i915.alpha_support=1 nouveau.modeset=0 iommu=no-igfx console=hvc0 acpi=off
ramdisk=initrd.img
```
**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**

https://www.qubes-os.org/doc/uefi-troubleshooting/
https://www.reddit.com/r/Qubes/comments/6f2kuu/tutorial_for_those_having_installationboot/
https://www.qubes-os.org/doc/intel-igfx-troubleshooting/
https://www.qubes-os.org/doc/nvidia-troubleshooting/
**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**
_A list of links to other bug reports, feature requests, or tasks in the qubes-issues tracker (or "none" if you didn't find any). Do not describe any other unreported bugs, features, or tasks here._
https://github.com/QubesOS/qubes-issues/issues/5074
The same system failed to boot XEN. The above issue resolved that bug. It also shows some debugging steps.

**Comment by JarrahG** [comment]
I have now tested the system with an installed version of Qubes, fully patched with both the standard dom0 kernel and kernel latest (5.3.6). Both of these result in the same kernel panic described above.

**Comment by JarrahG** [comment]
Searching online has found a few sources for similar kernel panics on CentOS Xen (mainly https://bugs.centos.org/view.php?id=6487). 

This recommended disabling cstates/power management in Xen/Linux and the BIOS. However, these also made no difference.

**Comment by JarrahG** [comment]
I'm out of ideas on this one. 
Dom0 kernel boots fine without Xen (lots of nouveau errors, but blacklisting nouveau solves those), as do other Linux distros (Specifically Fedora 30).

Most recent attempt was to disable the nvidia card at the PCI level, hiding it as if for PCI passthrough with:
```
xen-pciback.hide=(08:00.0)(08.00.1)
```

No change. Output exactly the same as above on the serial console. System hangs with no display on VGA console.

**Comment by JarrahG** [comment]
Tested on r4.1. System freezes shortly after bringing up CPUs and clocksource. No error output on serial console. 

Without serial console, system hangs with no output.

**Comment by marmarek** [comment]
EFI or legacy boot? I've seen a system with similar issue on legacy boot but working with EFI.

**Comment by JarrahG** [comment]
Interesting. This one is EFI only. There's no option for legacy boot.

**Comment by JarrahG** [comment]
This issue is resolved on the pre-release versions of Qubes 4.1
